adapters/repos/db/vector/hnsw/compress_segments_test.go

```diff
@@ -0,0 +1,85 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package hnsw
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/common"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
+	ssdhelpers "github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/testinghelpers"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+	ent "github.com/weaviate/weaviate/entities/vectorindex/hnsw"
+)
+
+func Test_NoRaceCompressAdaptsSegments(t *testing.T) {
+	efConstruction := 64
+	ef := 32
+	maxNeighbors := 32
+
+	dimensionsSet := []int{768, 125, 64, 27, 2, 19}
+	expectedSegmentsSet := []int{128, 25, 16, 9, 1, 19}
+	vectors_size := 1000
+
+	for i, dimensions := range dimensionsSet {
+		expectedSegments := expectedSegmentsSet[i]
+		vectors, _ := testinghelpers.RandomVecs(vectors_size, 1, dimensions)
+		distancer := distancer.NewL2SquaredProvider()
+
+		uc := ent.UserConfig{}
+		uc.MaxConnections = maxNeighbors
+		uc.EFConstruction = efConstruction
+		uc.EF = ef
+		uc.VectorCacheMaxObjects = 10e12
+		uc.PQ = ent.PQConfig{
+			Enabled: true,
+			Encoder: ent.PQEncoder{
+				Type:         "title",
```

Comment:
`tile`?

BTW, why 2 configs are needed? this one is provided to index constructor, but there is 2nd one used as an argument of `Compress` method

Reply:
I fixed the typo. About the second config, in auto pq there is no need for it. The old workflow was, you update the config and this triggers the compress. The two configs were not necessary the same. I think the TurnCompressionOn (in a separate PR) this was changed. There would not be any need for a second config in the future.